File: src/p0011/golang/solution.go
func maxArea(hs []int) int {
	result, i, j := 0, 0, len(hs)
	for i+1 < j {
		if hs[i] < hs[j-1] {
			if result < (j-1-i)*hs[i] {
				result = (j - 1 - i) * hs[i]
			}
			i++
		} else {
			if result < (j-1-i)*hs[j-1] {
				result = (j - 1 - i) * hs[j-1]
			}
			j--
		}
	}
	return result
}

File: src/p0011/rust/solution.rs
use std::cmp::max;

impl Solution {
    pub fn max_area(hs: Vec<i32>) -> i32 {
        let mut result: usize = 0;
        let mut i: usize = 0;
        let mut j: usize = hs.len();
        while i + 1 < j {
            if hs[i] < hs[j - 1] {
                result = max(result, (j - 1 - i) * hs[i] as usize);
                i += 1;
            } else {
                result = max(result, (j - 1 - i) * hs[j - 1] as usize);
                j -= 1;
            }
        }
        return result as i32;
    }
}
